android/sdk/src/main/java/com/taobao/weex/ui/component/helper/ScrollStartEndHelper.java

```diff
@@ -64,14 +67,16 @@ public void  onScrolled(int x, int y){
                 || component.getEvents().contains(Constants.Event.SCROLL_END))){
             this.x = x;
             this.y = y;
-            if(!hasStart){
+            if(!hasStart && canStart){
```

Comment:
This `if` block should be as code in line 95  fired the event.

Reply:
code inline 95 is for the case the list has reached the top or bottom that can not callback in onscroll

Reply:
Actually, you can safely delete code from line 95 to line 98,  which are never invoked in any case.